alert-common/src/main/java/com/synopsys/integration/alert/common/message/model/MessageContentGroup.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.message.model;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.commons.lang3.StringUtils;
+
+import com.synopsys.integration.alert.common.rest.model.AlertSerializableModel;
+
+public class MessageContentGroup extends AlertSerializableModel {
+    private final List<AggregateMessageContent> subContent;
+
+    private LinkableItem commonTopic;
+
+    public static MessageContentGroup singleton(final AggregateMessageContent message) {
```

Comment:
is this named singleton because it creating an actual Singleton? Or because this group will be for 1 message?
If it is for a single message then I think we should change the name to singleMessage or newSingleMessage

Reply:
For what it's worth, Java calls Collections with a single element singleton in their APIs. If that doesn't sway you I can change it. Singleton is a native java concept and distinct from the Singleton Pattern.